### DISCUSSION TITLE
Replicate window.open size setting with openURL

### DISCUSSION THREAD

**Discussion by maggie44** [quasarframework/quasar#14273] [General - Components / Directives / etc] [answered]
In dev mode on an Electron app I am trying to do the equivalent of `window.open(url, '_blank', 'width=1050')`. This line of code works, opening a new window at 1050. 

I can't seem to achieve the same with openURL though, the window size is always the same:

```
      openURL(
        url,
        mem(),
        {
          width: '2000',
        }
      )
```

I have tried `width: 2000` (without quotes) too.

```
Operating System - Darwin(21.6.0) - darwin/arm64
NodeJs - 16.16.0

Global packages
  NPM - 8.11.0
  yarn - 3.2.2
  @quasar/cli - undefined
  @quasar/icongenie - 2.5.3
  cordova - Not installed

Important local packages
  quasar - 2.7.7 -- Build high-performance VueJS user interfaces (SPA, PWA, SSR, Mobile and Desktop) in record time
  @quasar/app-vite - 1.0.6 -- Quasar Framework App CLI with Vite
  @quasar/extras - 1.15.1 -- Quasar Framework fonts, icons and animations
  eslint-plugin-quasar - Not installed
  vue - 3.2.37 -- The progressive JavaScript framework for building modern web UI.
  vue-router - 4.1.3
  pinia - 2.0.20 -- Intuitive, type safe and flexible Store for Vue
  vuex - Not installed
  vite - 2.9.15 -- Native-ESM powered web dev build tool
  eslint - 8.22.0 -- An AST-based pattern checker for JavaScript.
  electron - 20.0.3 -- Build cross platform desktop apps with JavaScript, HTML, and CSS
  electron-packager - 15.5.2 -- Customize and package your Electron app with OS-specific bundles (.app, .exe, etc.) via JS or CLI
  electron-builder - Not installed
  register-service-worker - 1.7.2 -- Script for registering service worker, with hooks
  @capacitor/core - Not installed
  @capacitor/cli - Not installed
  @capacitor/android - Not installed
  @capacitor/ios - Not installed

Quasar App Extensions
  *None installed*
```

**Comment by yusufkandemir** [comment, marked as answer]
Converted this into an issue: https://github.com/quasarframework/quasar/issues/14278, then fixed it in https://github.com/quasarframework/quasar/commit/15609bdaa90296b9b53e6c59f9004ebad6371ada. This will be included in the next `quasar` release.

**Reply by maggie44** [reply]
Thanks @yusufkandemir,

On a related issue, there seems to be an issue with the typings from open URL too:

```
      openURL(url, null, {
        width: '1050'
      })
```

`Argument of type 'null' is not assignable to parameter of type '((...args: any[]) => any) | undefined'.`

Yet the docs provide `null` as the example: https://quasar.dev/quasar-utils/other-utils#open-external-url

**Reply by maggie44** [reply]
> Thanks @yusufkandemir,
> 
> On a related issue, there seems to be an issue with the typings from open URL too:
> 
> ```
>       openURL(url, null, {
>         width: '1050'
>       })
> ```
> 
> `Argument of type 'null' is not assignable to parameter of type '((...args: any[]) => any) | undefined'.`
> 
> Yet the docs provide `null` as the example: https://quasar.dev/quasar-utils/other-utils#open-external-url

Probably a silly place to log this, I added it as an issue: https://github.com/quasarframework/quasar/issues/14292
